[MIG] product_pricelist_supplierinfo: Migration to 16.0

andreagidaltig opened the pull request.

cubells committed:
product_pricelist_supplierinfo: priceslists based in supplier info

pedrobaeza committed:
product_pricelist_supplierinfo: Several enhancements + concept fixing

* Don't depend on sales
* Extended README
* Tests focused on module specific features
* Code optimization
* Don't mix pricelist info with supplier info on criteria

pedrobaeza committed:
product_pricelist_supplierinfo: Take into account sales order date

Date order is passed by context, so we have to take this into account for
computing properly available supplierinfo records.

pedrobaeza committed:
product_pricelist_supplierinfo: Give more priority in order to min_qty

If not, you can twist the order in the form and rules won't be applied according criteria

Nikul-Chaudhary committed:
[MIG] product_pricelist_supplierinfo v11 to v12

carlosdauden committed:
[11.0][FIX] product_pricelist_supplierinfo: Add template compatibility (#436)

Previously, only worked fine with product.product.

pedrobaeza committed:
[FIX] product_pricelist_supplierinfo: Avoid false computations

If you set the value "Based on" for using supplier info, but then you change the
computation type to another one (like fixed price or discount), this code is still
acting, so we should check both fields.

pedrobaeza committed:
[FIX] product_pricelist_supplierinfo: Proper URL

eLBati committed:
Added translation using Weblate (Italian)

augustodinizl committed:
Added translation using Weblate (Portuguese (Brazil))

mourad-ehm committed:
[IMP] add margin on supplier info to configure the sale price

When having price per qty at the supplier level and if you want to
base your pricelist on the supplier price, now you can
define a margin on the product.supplierinfo

ruuter committed:
[FIX] product_pricelist_supplierinfo: use _select_seller

Use standard way to select the seller.

As a result a bug in previous solution is fixed also: before this,
all supplierinfo records having product_tmpl_id set, were considered
as possible price for all product variants, even if product_id was
set also. That resulted in same price for all variants even if
supplierinfo records had different price for all variants.

claudiagn committed:
Translated using Weblate (Spanish)

Currently translated at 93.3% (14 of 15 strings)

Translation: product-attribute-12.0/product-attribute-12.0-product_pricelist_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_pricelist_supplierinfo/es/

CarmenMiranda committed:
[FIX] product_pricelist_supplierinfo: Converting Price

If we have suppliers in two or more currencies, and we have a pricelist
with an item that is selected the "Prices based on supplier info" option
but the pricelist is in a currency different from the supplierinfo the
compute of the price get the price on the same amount but with a different
currency.
With these changes, we are converting the price to the currency on the
pricelist that is being used.
Also, a test was added to prove the previous behaviour.

eLBati committed:
IMP product_pricelist_supplierinfo allowing to filter suppliers to get prices from

ypapouin committed:
Added translation using Weblate (French)

ypapouin committed:
Translated using Weblate (French)

Currently translated at 100.0% (20 of 20 strings)

Translation: product-attribute-12.0/product-attribute-12.0-product_pricelist_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_pricelist_supplierinfo/fr/

ruuter committed:
[FIX] product_pricelist_supplierinfo: ensure product exists

It is possible that this method is called on an inactive template,
in which case product_variant_id returns empty recordset and
_select_seller will raise expected singleton error.

CarmenMiranda committed:
[FIX] product_pricelist_supplierinfo: Convert price to UOM on the Sale

If we have a product with a supplier with a UOM "A" but the product is
sold with a UOM "B" the price we are getting is the same no matter the
UOM. For example:
Product with the price 1200 per Dozen and we sell a 1 unit we are
getting the price of 1200 for that unit when we should be obtaining 100.

With these changes, we are converting the price to the UOM that is being
used on the sale or the default on the product.
Also, a test was added to prove the previous behaviour.

bosd committed:
Added translation using Weblate (Dutch)

claudiagn committed:
Added translation using Weblate (Catalan)

eLBati committed:
FIX product_pricelist_supplierinfo: _get_supplierinfo_pricelist_price when date is datetime.
date may come from context

CarlosRoca13 committed:
[IMP] product_pricelist_supplierinfo: black, isort

CarlosRoca13 committed:
[MIG] product_pricelist_supplierinfo: Migration to v13.0

[UPD] Update product_pricelist_supplierinfo.pot

[UPD] README.rst

CarlosRoca13 committed:
[FIX] product_pricelist_direct_print: Error when date is not defined

When we check the price of any product, the date that we pass on the context has to be a Datetime value or None. With this change we avoid to pass a False.

Here you have the reference to the code that check that https://github.com/OCA/OCB/blob/13.0/addons/product/models/product.py#L599-L600

TT29935

product_pricelist_supplierinfo 13.0.1.0.2

chienandalu committed:
[FIX] product_pricelist_supplierinfo: public users

In the e-commerce, if we have a pricelist with a rule based on
supplierinfo for a determined supplier partner, such partner could be
not accesible for public users and the affected products could not be
reached.

TT31476

victoralmau committed:
[IMP] product_pricelist_supplierinfo: black, isort, prettier

victoralmau committed:
[MIG] product_pricelist_supplierinfo: Migration to 14.0
TT31816

[UPD] Update product_pricelist_supplierinfo.pot

[UPD] README.rst

[IMP] update dotfiles [ci skip]

chienandalu committed:
[FIX] product_pricelist_supplierinfo: override min quantity

If we don't have a supplierinfo that below 1 unit we won't get any
seller. We want to ensure that the condition is fully ignored to get
whatever price given the other criterias.

TT33659

pedrobaeza committed:
[FIX] product_pricelist_supplierinfo: Allow to show shop for public users

0581b669ee435de4b1c068e6883e92a1e511ea41 is not enough on v13 for
avoiding the access error, so we sudoed the whole price fetch operation.

TT36898

victoralmau committed:
[MIG] product_pricelist_supplierinfo: Migration to 15.0

TT36975

victoralmau committed:
[FIX] product_pricelist_supplierinfo: Show text in the view

TT36975

product_pricelist_supplierinfo 14.0.1.1.0

Commit:
[UPD] Update product_pricelist_supplierinfo.pot

OCA-git-bot committed:
[UPD] README.rst

carlosdauden committed:
[IMP] product_pricelist_supplierinfo: New context force_filter_supplier_id

OCA-git-bot committed:
product_pricelist_supplierinfo 15.0.1.0.1

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 59.0% (13 of 22 strings)

Translation: product-attribute-15.0/product-attribute-15.0-product_pricelist_supplierinfo
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_pricelist_supplierinfo/it/

GuillemCForgeFlow committed:
[FIX]product_pricelist_supplierinfo: fix tests

OCA-git-bot committed:
product_pricelist_supplierinfo 15.0.1.0.2

andreagidaltig committed:
[IMP] product_pricelist_supplierinfo: pre-commit stuff

andreagidaltig commented:
Hi @luisg123v could you review this please?

luisg123v commented:
/ocabot migration product_pricelist_supplierinfo

luisg123v commented on product_pricelist_supplierinfo/__manifest__.py:
```diff
@@ -4,7 +4,7 @@
     "name": "Supplier info prices in sales pricelists",
     "summary": "Allows to create priceslists based on supplier info",
-    "version": "15.0.1.0.2",
+    "version": "16.0.1.0.0",
```
This should go in the migration commit, not the pre-commit commit

luisg123v commented on product_pricelist_supplierinfo/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+    "version": "16.0.1.0.0",
+    "category": "Sales",
+    "website": "https://github.com/OCA/product-attribute",
+    "author": "Vauxoo, Tecnativa, Odoo Community Association (OCA)",
```
Add it at the end

luisg123v commented on product_pricelist_supplierinfo/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+    "category": "Sales",
+    "website": "https://github.com/OCA/product-attribute",
+    "author": "Vauxoo, Tecnativa, Odoo Community Association (OCA)",
+    "manteiners": ["luisg123v"],
```
```suggestion
    "maintainers": ["luisg123v"],
```

luisg123v commented on product_pricelist_supplierinfo/models/product_pricelist.py:
```diff
@@ -0,0 +1,48 @@
+        getting proper prices when based on supplier info.
+        """
+        rule_obj = self.env["product.pricelist.item"]
+        result = super()._compute_price_rule(products, qty, uom, date)
```
You need to pass all params

```suggestion
        result = super()._compute_price_rule(products, qty, uom, date, **kwargs)
```

luisg123v commented on product_pricelist_supplierinfo/readme/CONTRIBUTORS.rst:
```diff
@@ -0,0 +1,13 @@
+
+  * Lorenzo Battistini
+
+* Andrea Gidalti <andreag@vauxoo.com> https://vauxoo.com/
```
Remove URL

luisg123v commented on product_pricelist_supplierinfo/tests/test_product_supplierinfo.py:
```diff
@@ -0,0 +1,332 @@
+
+from datetime import date
+
+from odoo.tests import common, tagged
```
```suggestion
from odoo.tests import TransactionCase, tagged
```

luisg123v commented on product_pricelist_supplierinfo/tests/test_product_supplierinfo.py:
```diff
@@ -0,0 +1,332 @@
+            {
+                "name": "Product Test",
+                "seller_ids": [
+                    (0, 0, {"partner_id": cls.supplier1.id, "min_qty": 5, "price": 50}),
```
Prefer the Command syntax

Same below

luisg123v commented on product_pricelist_supplierinfo/tests/test_product_supplierinfo.py:
```diff
@@ -0,0 +1,332 @@
+            self.product.product_tmpl_id.with_context(
+                pricelist=self.pricelist.id
+            )._get_contextual_price(),
+            10.0,
```
Remove this assertion, this is not required anymore. It was to check value was propagated to the pricelist field and, since such field doesn't exist anymore, there's no point on checking it.

Same in all cases below.

It would be a good idea to document this in commit description

luisg123v replied:
I see you documented this in commit description, you just need to split lines to 72 characters.

On the other hand, I'd suggest something more clear, e.g.
```
Assertion in tests involving the pricelist field in products were
removed, asthat field doesn't exist anymore [1], and such assertions
were intended only to ensure computed price were propagated correctly to
that field.

[1] https://github.com/odoo/odoo/commit/9e99a9df
```

moylop260 commented:
/ocabot migration product_pricelist_supplierinfo

luisg123v commented on product_pricelist_supplierinfo/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+    "name": "Supplier info prices in sales pricelists",
+    "summary": "Allows to create priceslists based on supplier info",
+    "version": "16.0.1.0.0",
+    "category": "Sales",
```
```suggestion
    "category": "Sales/Sales",
```

luisg123v commented on product_pricelist_supplierinfo/tests/test_product_supplierinfo.py:
```diff
@@ -0,0 +1,315 @@
+                                    ).id,
+                                ),
+                                (
+                                    4,
```
Command syntax is missing here and below

andreagidaltig committed:
[MIG] product_pricelist_supplierinfo: Migration to 16.0

Assertion in tests involving the pricelist field in products were
removed, as that field doesn't exist anymore [1], and such assertions
were intended only to ensure computed price were propagated correctly to
that field.

[1] https://github.com/odoo/odoo/commit/9e99a9df

luisg123v reviewed (approved):
LGTM :+1: 

Hi @moylop260,

Could you review, please?

moylop260 commented:
/ocabot merge nobump
